PyNite/Member3D.py

```diff
@@ -1186,6 +1186,92 @@ def PlotDeflection(self, Direction):
         Member3D.__plt.xlabel('Location')
         Member3D.__plt.title('Member ' + self.Name)
         Member3D.__plt.show()
+    
+#%%   
+    def RelativeDeflection(self, Direction, x):
+        
+        d = self.d()
+        d1y = d[1,0]
+        d2y = d[7,0]
+        d1z = d[2,0]
+        d2z = d[8,0]
+        L = self.L()
+
+        """
+        Returns the relative deflection at a point along the member's length
+        
+        Parameters
+        ----------
+        Direction : string
+            The direction in which to find the relative deflection. Must be one of the following:
+                "dy" = Deflection in the local y-axis
+                "dz" = Deflection in the local x-axis
+        x : number
+            The location at which to find the relative deflection
+        """
+        
+        # Check which axis is of interest
+        if Direction == "dy":
+            
+            # Check which segment "x" falls on
+            for segment in self.SegmentsZ:
+                
+                if round(x,10) >= round(segment.x1,10) and round(x,10) < round(segment.x2,10):
+                    
+                    return (segment.Deflection(x - segment.x1)) - (d1y + (d2y-d1y)/L *x)
+                
+            if isclose(x, self.L()):
+                
+                lastIndex = len(self.SegmentsZ) - 1
+                return (self.SegmentsZ[lastIndex].Deflection(x - self.SegmentsZ[lastIndex].x1))-d2y
+                
+        elif Direction == "dz":
+            
+            for segment in self.SegmentsY:
+                
+                if round(x,10) >= round(segment.x1,10) and round(x,10) < round(segment.x2,10):
+                    
+                    return (segment.Deflection(x - segment.x1)) - (d1z + (d2z-d1z)/L *x)
+                
+            if isclose(x, self.L()):
+                
+                lastIndex = len(self.SegmentsY) - 1
+                return (self.SegmentsY[lastIndex].Deflection(x - self.SegmentsY[lastIndex].x1)) - d2z
+
+#%%
+    def PlotRelativeDeflection(self, Direction):
+        """
+        Plots the deflection diagram for the member
+        
+        Parameters
+        ----------
+        Direction : {"dy", "dz"}
+            The direction in which to plot the deflection.
+        """
+                
+        # Import 'pyplot' if not already done
+        if Member3D.__plt is None:
+            from matplotlib import pyplot as plt
+            Member3D.__plt = plt
+        
+        fig, ax = Member3D.__plt.subplots()
+        ax.axhline(0, color='black', lw=1)
+        ax.grid()
+        
+        x = []
+        d_relative = []
+        
+        # Calculate the relative deflection diagram
+        for i in range(20):
+            
+            x.append(self.L() / 19 * i)
```

Comment:
```suggestion
            x.append(self.L()/19*i)
```
This follows the PEP8 python style guide. Spaces in mathematics should only be between `+` and `-` signs.